plugin/arcconf/arcconf.py

```diff
@@ -543,44 +543,45 @@ def disks(self, search_key=None, search_value=None,
+    def volume_raid_create(self, name, raid_type, disks, strip_size, flags=Client.FLAG_RSVD):
+        """
+
+        :param name: name of volume (this will be ignored)
+        :param raid_type: RAID Type of the LD
+        :param disks: Physical device
+        :param strip_size: Strip size
+        :param flags: for future use
+        :return: volume object of newly created volume
+
+        Depends on command:
+            arcconf create <ctrlNo> logicaldrive max <RAID LEVEL> <channel ID> <Device ID> <channel ID> <Device ID> \
+            <channel ID> <Device ID>...
+        """
 
         arcconf_raid_level = _lsm_raid_type_to_arcconf(raid_type)
         arcconf_disk_ids = []
         ctrl_num = None
-        disk_channel = ''
-        disk_device = ''
-        disk_dict = {'Channel': '0', 'Device': '1'}
-        lsm_vols = []
+        disk_dict = {'Channel': '0', 'Device': '0'}
 
         for disk in disks:
             if not disk.plugin_data:
-                raise LsmError(
-                    ErrorNumber.INVALID_ARGUMENT,
-                    "Illegal input disks argument: missing plugin_data "
+                raise LsmError(ErrorNumber.INVALID_ARGUMENT, "Illegal input disks argument: missing plugin_data "
                     "property")
-            (cur_ctrl_num, disk_channel, disk_device) = \
-                disk.plugin_data.split(',')[:3]
-
-            requested_disks = [d.name for d in disks]
-            for disk_name in requested_disks:
-                if str(disk_name) == str(disk.name):
-                    disk_channel = str(disk_channel.strip())
-                    disk_device = str(disk_device.strip())
-                    disk_dict.update(
-                        {'Channel': disk_channel, 'Device': disk_device})
-                    arcconf_disk_ids.append(disk_dict.copy())
-                    disk_dict = {}
-                    if ctrl_num is None:
-                        ctrl_num = cur_ctrl_num
-                    elif ctrl_num != cur_ctrl_num:
-                        raise LsmError(
-                            ErrorNumber.INVALID_ARGUMENT,
-                            "Illegal input disks argument: disks "
-                            "are not from the same controller/system.")
-
-        cmds = ["create", ctrl_num, "logicaldrive", "1024", arcconf_raid_level]
+            (cur_ctrl_num, disk_channel, disk_device) = disk.plugin_data.split(',')[:3]
+
+            if ctrl_num is None:
+                ctrl_num = cur_ctrl_num
+            elif ctrl_num != cur_ctrl_num:
+                raise LsmError(ErrorNumber.INVALID_ARGUMENT, "Illegal input disks argument: disks "
+                    "are not from the same controller/system.")
+
+            disk_channel = str(disk_channel.strip())
+            disk_device = str(disk_device.strip())
+            disk_dict.update({'Channel': disk_channel, 'Device': disk_device})
+            arcconf_disk_ids.append(disk_dict.copy())
```

Comment:
Why use the `disk_dict`  when you could simply use:

```python
arcconf_disk_ids.append({'Channel': disk_channel, 'Device': disk_device})
```

Reply:
done - commit id - ab54bc5b54a37fedd78c6232d19408f1ea78c3a8